src/digraph.rs

```diff
@@ -1073,6 +1073,45 @@ impl PyDiGraph {
         Ok(out_map)
     }
 
+    /// Get the neighbors of a node.
+    ///
+    /// This with return a list of neighbor node indices
+    ///
+    /// :param int node: The index of the node to get the neibhors of
+    ///
+    /// :returns: A list of the neighbor node indicies
+    /// :rtype: list
+    pub fn neighbors(&self, node: usize) -> Vec<usize> {
+        self.graph
+            .neighbors(NodeIndex::new(node))
+            .map(|node| node.index())
+            .collect()
+    }
+
+    /// Get the directed neighbors of a node.
+    ///
+    /// This with return a list of directed neighbor node indices
+    ///
+    /// :param int node: The index of the node to get the neibhors of
+    ///
+    /// :returns: A list of the neighbor node indicies
+    /// :rtype: list
+    pub fn neighbors_directed(
```

Comment:
That's a good point, I will change it to be `successor_indices` and `predecessor_indices` those are better function names. The use case here specifically was for: https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/transpiler/passes/layout/noise_adaptive_layout.py#L124-L126 where the nodes don't have a payload/weight, so the `PyObject` is `None`. We need to return the the indices so that pass can do a lookup on the graph matrices there.

The overhead for returning a list of `PyObject`s isn't actually that high since a `PyObject` is basically just a pointer/reference to an object in the Python heap (https://pyo3.rs/master/types.html#pyt-and-pyobject ). The issue is that there really isn't a mapping from `PyObject` -> graph node (for a qiskit `DAGCircuit` we have this because the weight `PyObject`, is a `DAGNode` object which contains a node_id attribute).

Reply:
Ok, the function was split and renamed in: https://github.com/Qiskit/retworkx/pull/147/commits/75da922d4da7c0444b391a8c4cedc10344b5f49f

Reply:
I understand in what case current `successors` (returning PyObject) is not sufficient, i.e. why we need `successor_indices`. Thank you.